test_data/pil/sum_via_witness_query.pil

```diff
@@ -7,7 +7,13 @@ namespace Sum(%N);
     pol fixed ISALMOSTLAST(i) { if i == %last_row - 1 { 1 } else { 0 } };
     pol fixed ISFIRST = [ 1, 0 ] + [0]*;
 
-    col witness input(i) query ("input", i);
```

Comment:
This used to ask for prover input 3 in row 3, but we (intentionally) only provide 3 inputs, so it returns `Error accessing prover inputs: Index 3 out of bounds 3`.

Previously, this was fine, because when processing the `(N - 1, 0)` row pair, we would "see" the `ISLAST * input = 0;` identity first, so the query was never run. Now, since this identity is skipped, the query is run and returns an error.

I think asking for a not provided error *should* return an error. Actually, whether an error is returned or `None` is up to the query callback, but our default one returns an error. I changed the query so now it returns `None` in row 3 regardless of the query callback.